### PR TITLE
Issue with the interest: restricted duplicate interest create for same organisation

### PR DESCRIPTION
**Pull request by ruksana2808**

### CODE REVIEW

File: src/main/java/com/igot/cb/interest/service/impl/InterestServiceImpl.java [hunk 1]
```diff
@@ -120,6 +120,22 @@ public CustomResponse createInterest(JsonNode interestDetails) {
         JsonNode fetchedDemandJson = demandEntity.get().getData();
         ((ObjectNode) fetchedDemandJson).put(Constants.INTEREST_COUNT,
```
Comment (Sahil-tarento) [inline src/main/java/com/igot/cb/interest/service/impl/InterestServiceImpl.java:129]: Please add check to check null pointer

File: src/main/java/com/igot/cb/interest/service/impl/InterestServiceImpl.java [hunk 2]
```diff
@@ -120,6 +120,22 @@ public CustomResponse createInterest(JsonNode interestDetails) {
...
             fetchedDemandJson.get(Constants.INTEREST_COUNT).asInt() + 1);
+        Set<String> interestOrgSet = new HashSet<>();
+        if (fetchedDemandJson.has(Constants.INTEREST_ORG_SET) && !ObjectUtils.isEmpty(fetchedDemandJson.get(Constants.INTEREST_ORG_SET))) {
+          for (JsonNode node : fetchedDemandJson.get(Constants.INTEREST_ORG_SET)) {
+            interestOrgSet.add(node.asText());
+          }
+        }
+        String orgId = interestDetails.get(Constants.ORG_ID).asText();
```
Comment (Sahil-tarento) [inline src/main/java/com/igot/cb/interest/service/impl/InterestServiceImpl.java]: Just check null check

**Review by Sahil-tarento — CHANGES_REQUESTED** [review]

File: src/main/java/com/igot/cb/interest/service/impl/InterestServiceImpl.java (hunk 1 above)
Comment (ruksana2808) [reply src/main/java/com/igot/cb/interest/service/impl/InterestServiceImpl.java:129]: Made changes

File: src/main/java/com/igot/cb/interest/service/impl/InterestServiceImpl.java (hunk 2 above)
Comment (ruksana2808) [reply src/main/java/com/igot/cb/interest/service/impl/InterestServiceImpl.java]: orgId is a mandatory field so in the validation method itself we are checking whether it is empty , null or missing

**Pull request merged**